test/parallel/test-buffer-fill.js

```diff
@@ -406,3 +394,38 @@ assert.throws(() => {
   });
   buf.fill('');
 }, /^RangeError: out of range index$/);
+
+assert.deepStrictEqual(
+    Buffer.allocUnsafeSlow(16).fill('ab', 'utf16le'),
+    Buffer.from('61006200610062006100620061006200', 'hex'));
+
+assert.deepStrictEqual(
+    Buffer.allocUnsafeSlow(15).fill('ab', 'utf16le'),
+    Buffer.from('610062006100620061006200610062', 'hex'));
+
+assert.deepStrictEqual(
+    Buffer.allocUnsafeSlow(16).fill('ab', 'utf16le'),
+    Buffer.from('61006200610062006100620061006200', 'hex'));
+assert.deepStrictEqual(
+    Buffer.allocUnsafeSlow(16).fill('a', 'utf16le'),
+    Buffer.from('61006100610061006100610061006100', 'hex'));
+
+assert.strictEqual(
+    Buffer.allocUnsafeSlow(16).fill('a', 'utf16le').toString('utf16le'),
+    'aaaaaaaa');
+assert.strictEqual(
+    Buffer.allocUnsafeSlow(16).fill('a', 'latin1').toString('latin1'),
+    'aaaaaaaaaaaaaaaa');
+assert.strictEqual(
+    Buffer.allocUnsafeSlow(16).fill('a', 'utf8').toString('utf8'),
+    'aaaaaaaaaaaaaaaa');
+
+assert.strictEqual(
+    Buffer.allocUnsafeSlow(16).fill('Љ', 'utf16le').toString('utf16le'),
+    'ЉЉЉЉЉЉЉЉ');
+assert.strictEqual(
+    Buffer.allocUnsafeSlow(16).fill('Љ', 'latin1').toString('latin1'),
+    '\t'.repeat(16));
```

Comment:
Suggestion: maybe use `String#repeat()` for the other strings as well, it looks more Obviously Correct.